torch/csrc/autograd/custom_function.h

```diff
@@ -205,6 +205,12 @@ struct ExtractVariables : IterArgs<ExtractVariables> {
     is_var_.push_back(true);
     list_.emplace_back(x);
   }
+  void operator()(const at::TensorList& list) {
+    for (const at::Tensor& x : list) {
+      is_var_.push_back(true);
+      list_.emplace_back(x);
```

Comment:
@albanD @soulitzer I want check with you before approving.

This sounds reasonable (adding support for tensorlist arguments with autograd::functions) - was this just an API limitation that we never needed until now? Or are there any reasons we didn't add this sooner.

Reply:
One thing this does is diverge the behavior between the python and c++ custom Functions. In python, tensors in lists are interpreted as not having gradients.

Maybe this is reasonable to do though because in python the workaround is easy (just pass in `*args`) but I'm not as sure about what the workaround would be in c++.

Reply:
I'm not sure of any other reasons not to have this, maybe @albanD has more context (he is on PTO until next week though).

Reply:
> In python, tensors in lists are interpreted as not having gradients

We need this precisely because we need to support gradients for TensorList.
https://github.com/pytorch/xla/pull/3843

Reply:
Thanks @soulitzer , this will unblock one of important pr for pytorch/xla so I am wondering if it is OK to give it a green light if you don't see any problem for now. We can revisit and work on it more if Alban has other comments.

Reply:
Thanks for the context, sounds good to me!